src/erasure-code/isa/ErasureCodeIsa.cc

```diff
@@ -114,29 +114,111 @@ int ErasureCodeIsa::decode_chunks(const set<int> &want_to_read,
   return isa_decode(erasures, data, coding, blocksize);
 }
 
+void ErasureCodeIsaDefault::encode_delta(const bufferptr &old_data,
+                                  const bufferptr &new_data,
+                                  bufferptr *delta)
+{
+  
+  //TODO handle old_data or new_data being a special zero buffer.
+  
+  constexpr int data_vectors = 2;
+  char * data[data_vectors];
+  data[0] = const_cast<char*>(old_data.c_str());
+  data[1] = const_cast<char*>(new_data.c_str());
+  char * coding = delta->c_str();
+  //TODO we need to be able to handle the delta buffer being the same
+  // buffer as the old_data buffer. Call isa_xor_with_new_coding_buf
+  isa_xor(data, &coding, delta->length(), data_vectors);
+  
+  //TODO return a special zero buffer if delta is all zeroes
+}
+
+void ErasureCodeIsaDefault::apply_delta(const map<int, bufferptr> &in,
+                                        map <int, bufferptr> &out)
+{
+  auto first = in.begin();
+  const unsigned blocksize = first->second.length();
+
+  //for each delta chunk in the in map
+  for (auto i = in.begin(); i != in.end(); i++) {
```

Comment:
Nit: Could use range based for loop:

`for (auto i : in) {
`
Could make it even more readable with:

`for (auto const& [datashard, databl] : in) {
`

And then use datashard instead of i->first and databl instead of i->second